lib/services/server.go

```diff
@@ -97,19 +97,19 @@ func compareServers(a, b types.Server) int {
 	if !cmp.Equal(a.GetCmdLabels(), b.GetCmdLabels()) {
 		return Different
 	}
-	if !a.Expiry().Equal(b.Expiry()) {
-		return OnlyTimestampsDifferent
-	}
 	if a.GetTeleportVersion() != b.GetTeleportVersion() {
 		return Different
 	}
-
 	if !cmp.Equal(a.GetApps(), b.GetApps()) {
 		return Different
 	}
 	if !cmp.Equal(a.GetKubernetesClusters(), b.GetKubernetesClusters()) {
 		return Different
 	}
+	// This check should always be last.
+	if !a.Expiry().Equal(b.Expiry()) {
```

Comment:
Maybe we should explain why as well?
```suggestion
	// OnlyTimestampsDifferent check must be after all Different checks.
	if !a.Expiry().Equal(b.Expiry()) {
```

Reply:
Yes that is much clearer updated here [78df4b1](https://github.com/gravitational/teleport/pull/10380/commits/78df4b18a03667b2240d5b975cb0a6bd86b60f81)